modules/core/include/visp3/core/vpArray2D.h

```diff
@@ -308,18 +310,76 @@ template <class Type> class vpArray2D
   }
 #endif
 
+  /**
+   * \brief Creates a view of the Matrix A.
+   * A view shares the same underlying memory as the original array.
+   * It can be written into, modifying the original data. However, the array cannot be resized.
+   *
+   * When you use this method, it is your responsibility to ensure that the lifespan of the view does not exceed the lifespan of the original array.
+   *
+   * \param A the array to view
+   * \return vpArray2D<T>
+   */
+  static vpArray2D<Type> view(const vpArray2D<Type> &A)
+  {
+    vpArray2D<Type> v;
+    v.rowNum = A.rowNum;
+    v.colNum = A.colNum;
+    v.rowPtrs = A.rowPtrs;
+    v.dsize = A.dsize;
+    v.data = A.data;
+    v.isMemoryOwner = false;
+    v.isRowPtrsOwner = false;
+    return v;
+  }
+  /**
+   * @brief Create an array view of a raw data.
+   * This data is not owned by the resulting array and should be freed after the array is destroyed (not before)
+   *
+   * @param data Pointer to the raw data
+   * @param numRows Number of rows
+   * @param numCols Number of columns
+   * @return vpArray2D<Type>
+   */
+  static vpArray2D<Type> view(Type *data, unsigned int numRows, unsigned int numCols)
+  {
+    vpArray2D<Type> v;
+    vpArray2D<Type>::view(v, data, numRows, numCols);
+    return v;
+  }
+
+  static void view(vpArray2D<Type> &v, Type *data, unsigned int numRows, unsigned int numCols)
+  {
+    v.rowNum = numRows;
+    v.colNum = numCols;
+    v.dsize = numRows * numCols;
+    v.data = data;
+    v.isMemoryOwner = false;
+    v.isRowPtrsOwner = true;
+    v.rowPtrs = reinterpret_cast<Type **>(malloc(v.rowNum * sizeof(Type *)));
```

Comment:
@SamFlt Please document this function.
Here we should add a security test for `rowPtrs`
```
if ((v.isRowPtrsOwner == true) && (v.rowPtrs != nullptr)) {
  free(v.rowPtrs);
}
v.rowPtrs = reinterpret_cast<Type **>(malloc(v.rowNum * sizeof(Type *)));
```